Obleak.Fody/ObleakWeaver.cs

```diff
@@ -0,0 +1,183 @@
+﻿using System;
+using System.Linq;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Mono.Cecil.Rocks;
+
+namespace Obleak.Fody
+{
+    public class ObleakWeaver
+    {
+        private const string COMPOSITE_DISPOSABLE_FIELD_NAME = "$ObleakCompositeDisposable";
+
+        private readonly Func<Instruction, TypeReference, string, bool> _isExpectedMethodCall =
+            (instruction, expectedReturnType, expectedName) =>
+                instruction?.Operand != null &&
+                (instruction.OpCode == OpCodes.Call || instruction.OpCode == OpCodes.Callvirt) &&
+                instruction.Operand is MethodReference &&
+                ((MethodReference)instruction.Operand).ReturnType.FullName == expectedReturnType.FullName &&
+                ((MethodReference)instruction.Operand).FullName.Contains(expectedName);
+
+        public ModuleDefinition ModuleDefinition { get; set; }
+
+        // Will log an MessageImportance.High message to MSBuild. OPTIONAL
+        public Action<string> LogInfo { get; set; }
+
+        // Will log an error message to MSBuild. OPTIONAL
+        public Action<string> LogError { get; set; }
+
+        public void Execute()
+        {
+            LogInfo("Obleak weaving");
+
+            // Validate we have the dependencies we need
+            var rxCore = ModuleDefinition.AssemblyReferences.Where(x => x.Name == "System.Reactive.Core").OrderByDescending(x => x.Version).FirstOrDefault();
```

Comment:
Is it worth pulling out this to a func? The logic for 'get latest version of named reference or null' is repeated thrice in this method.